src/components/views/messages/TimelineSeparator.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+
+interface Props {
+    label: string;
+}
+
+export const enum SeparatorKind {
+    None,
+    Date,
+    LateEvent,
+}
+
+const TimelineSeparator: React.FC<Props> = ({ label, children }) => {
```

Comment:
Could we add a JSDoc comment to describe this component. And similarly in `DateSeparator` to explain why we have a more specialised component.